test/core/image_processing/threshold_truncate.cpp

```diff
@@ -18,101 +18,210 @@ int main()
 {
     int height = 4;
     int width = 4;
-    gil::gray8_image_t original(width, height), threshold(width, height),
-        expected(width, height);
+
+
+    /*---------------------------------------------*/
+    /*                                             */
+    /*                Gray to Gray                 */
+    /*                                             */
+    /*---------------------------------------------*/
```

Comment:
The comment separators are not the best choice to structure the code. 
Simply, extract each test case as a regular function and call from the `main`, like so:

https://github.com/boostorg/gil/blob/4ed7701b47764edf327e7c82f65cf9815026bdc2/test/core/image_view/collection.cpp#L93-L97

For example:

```
binary_gray_to_gray
binary_inverse_gray_to_gray
truncate_threshold_regular_gray_to_gray
truncate_zero_regular_gray_to_gray
```
etc.

At one point, especially after making I/O PNM a built-in format, https://lists.boost.org/boost-gil/2019/07/0255.php, we may decide to reformat the tests from the Boost.LightweightTest to regular Boost.Test.

Diving into Boost.Test may be too distracting for you, so I don't expect you do be doing that.